File: content/snippets/python/interview_lcci_01_03_string_to_url.py
def urlify(s: str, true_length: int) -> str:
    return "".join("%20" if ch == " " else ch for ch in s[:true_length])


def main() -> None:
    print(urlify("Mr John Smith    ", 13))


if __name__ == "__main__":
    main()

File: content/snippets/go/interview_lcci_01_03_string_to_url.go
package main

import (
	"fmt"
	"strings"
)

func urlify(s string, trueLength int) string {
	var b strings.Builder
	for i := 0; i < trueLength; i++ {
		if s[i] == ' ' {
			b.WriteString("%20")
		} else {
			b.WriteByte(s[i])
		}
	}
	return b.String()
}

func main() {
	fmt.Println(urlify("Mr John Smith    ", 13))
}
